### PR TITLE
chore(linter): enable errcheck and ineffassign linters

### PR DESCRIPTION
**Pull request by mangalaman93**

### CODE REVIEW

**Review by skrdgraph — APPROVED** [review]

**Comment by skrdgraph** [comment]
On the CI run .. this seems to have no issues.

I am checking locally to make sure this doesn't break after merge (like last time).

Screen Shot 2023-02-03 at 05 13 12

**Comment by skrdgraph** [comment]
We should not merge this, there is something wrong with this when run locally: 

```
Sudhishs-MacBook-Pro-Work:dgraph sudhishkr$ golangci-lint run


xidmap/trie.go:76:15: Error return value of `t.buf.Release` is not checked (errcheck)
        t.buf.Release()
                     ^
dgraph/cmd/live/run.go:191:19: Error return value of `txn.Discard` is not checked (errcheck)
        defer txn.Discard(ctx)
                         ^
dgraph/cmd/live/run.go:667:19: Error return value of `txn.Discard` is not checked (errcheck)
        defer txn.Discard(ctx)
                         ^
worker/backup_ee.go:327:17: Error return value of `th.buf.Release` is not checked (errcheck)
                th.buf.Release()
                              ^
worker/predicate_move.go:169:19: Error return value of `buf.SliceIterate` is not checked (errcheck)
                buf.SliceIterate(func(_ []byte) error {
                                ^
worker/predicate_move.go:289:20: Error return value of `buf.Release` is not checked (errcheck)
                defer buf.Release()
                                 ^
worker/restore_map.go:166:19: Error return value of `buf.Release` is not checked (errcheck)
        defer buf.Release()
                         ^
worker/restore_map.go:180:18: Error return value of `buf.SliceIterate` is not checked (errcheck)
        buf.SliceIterate(func(slice []byte) error {
                        ^
worker/restore_map.go:281:19: Error return value of `buf.Release` is not checked (errcheck)
        defer buf.Release()
                         ^
worker/restore_map.go:437:25: Error return value of `req.lbuf.Release` is not checked (errcheck)
                defer req.lbuf.Release()
                                      ^
worker/restore_reduce.go:255:17: Error return value of `cbuf.Release` is not checked (errcheck)
                                cbuf.Release()
                                            ^
worker/restore_reduce.go:270:15: Error return value of `cbuf.Release` is not checked (errcheck)
                cbuf.Release()
                            ^
worker/restore_reduce.go:284:16: Error return value of `kvBuf.Release` is not checked (errcheck)
                kvBuf.Release()
                             ^
worker/restore_reduce.go:311:15: Error return value of `cbuf.Release` is not checked (errcheck)
                cbuf.Release()
                            ^
dgraph/cmd/bulk/count_index.go:142:22: Error return value of `outBuf.Release` is not checked (errcheck)
        defer outBuf.Release()
                            ^
dgraph/cmd/bulk/count_index.go:166:18: Error return value of `buf.SliceIterate` is not checked (errcheck)
        buf.SliceIterate(func(slice []byte) error {
                        ^
dgraph/cmd/bulk/count_index.go:184:21: Error return value of `c.countBuf.Release` is not checked (errcheck)
                c.countBuf.Release()
                                  ^
dgraph/cmd/bulk/mapper.go:186:19: Error return value of `cbuf.SliceIterate` is not checked (errcheck)
        cbuf.SliceIterate(func(slice []byte) error {
                         ^
dgraph/cmd/bulk/mapper.go:284:19: Error return value of `sh.cbuf.Release` is not checked (errcheck)
                        sh.cbuf.Release()
                                       ^
dgraph/cmd/bulk/reduce.go:323:29: Error return value of `req.countBuf.Release` is not checked (errcheck)
                defer req.countBuf.Release()
                                          ^
dgraph/cmd/bulk/reduce.go:332:28: Error return value of `req.countBuf.SliceIterate` is not checked (errcheck)
                req.countBuf.SliceIterate(func(slice []byte) error {
                                         ^
dgraph/cmd/bulk/reduce.go:362:21: Error return value of `ci.writer.Write` is not checked (errcheck)
                                        ci.writer.Write(buf)
                                                       ^
dgraph/cmd/bulk/reduce.go:369:17: Error return value of `kvBuf.Release` is not checked (errcheck)
                        kvBuf.Release()
                                     ^
dgraph/cmd/bulk/reduce.go:423:19: Error return value of `cbuf.SliceIterate` is not checked (errcheck)
        cbuf.SliceIterate(func(slice []byte) error {
                         ^
dgraph/cmd/bulk/reduce.go:713:16: Error return value of `kvBuf.Release` is not checked (errcheck)
                kvBuf.Release()
                             ^
edgraph/server.go:1376:17: Error return value of `grpc.SendHeader` is not checked (errcheck)
        grpc.SendHeader(ctx, md)
                       ^
query/outputnode.go:961:6: ineffectual assignment to slice (ineffassign)
                                        slice = cur
                                        ^
query/outputnode.go:972:4: ineffectual assignment to slice (ineffassign)
                        slice = nil
                        ^
dgraph/cmd/bulk/reduce.go:604:3: ineffectual assignment to slice (ineffassign)
                slice, next := []byte{}, start
                ^
dgraph/cmd/live/run.go:614:6: ineffectual assignment to tlsConfig (ineffassign)
        var tlsConfig *tls.Config = nil
            ^
```

**Comment by skrdgraph** [comment]
convert to draft

**Comment by skrdgraph** [comment]
converting this to draft* (its still a WIP)

File: graphql/admin/http.go
Comment (mangalaman93) [inline graphql/admin/http.go:284]: same here

**Comment by skrdgraph** [comment]
Just validated a local run - this looks good. Looks like the `golangci` ran fine. This should hold.

**Review by skrdgraph — APPROVED** [review]
👍

File: contrib/integration/acctupsert/main.go
Comment (matthewmcneely) [inline contrib/integration/acctupsert/main.go:146]: Nit: Log entries should be capitalized

File: dgraph/cmd/bulk/count_index.go
Comment (matthewmcneely) [inline dgraph/cmd/bulk/count_index.go:117]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/count_index.go
Comment (matthewmcneely) [inline dgraph/cmd/bulk/count_index.go:149]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/count_index.go
Comment (matthewmcneely) [inline dgraph/cmd/bulk/count_index.go:184]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/count_index.go
Comment (matthewmcneely) [inline dgraph/cmd/bulk/count_index.go:197]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/mapper.go
Comment (matthewmcneely) [inline dgraph/cmd/bulk/mapper.go:160]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/mapper.go
Comment (matthewmcneely) [inline dgraph/cmd/bulk/mapper.go:291]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/reduce.go
Comment (matthewmcneely) [inline dgraph/cmd/bulk/reduce.go:325]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/reduce.go
Comment (matthewmcneely) [inline dgraph/cmd/bulk/reduce.go:341]: Nit: First word of log output should be capitalized

File: graphql/admin/namespace.go
Comment (matthewmcneely) [inline graphql/admin/namespace.go:70]: Error messages should not be capitalized

**Review by harshil-goel — APPROVED** [review]

**Review by matthewmcneely — CHANGES_REQUESTED** [review]
Nit: There are dozens of places in this PR where the first word of log output is not capitalized. The Go conventions we follow require the opposite. I realize this isn't super-consistent across the code-base, but... (note: I gave up commenting on each occurrence after 9 or so)

**Comment by mangalaman93** [comment]
@matthewmcneely I think it is the opposite. Everything should start from small letters, nothing should be capitalized.

**Comment by matthewmcneely** [comment]
@mangalaman93 Well it's definitely inconsistent throughout. But IIRC, the convention is that the first word in output to logs *should* be capitalized. Error messages should NEVER be capitalized or punctutated, because they are likely to be chained together.

I guess we just need to declare which way is "the way" for Dgraph.

**Pull request merged**
